src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.utility.OperationHandlerSwitch;
+import core.basesyntax.utility.impl.OperationHandlerSwitchImpl;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    @Override
+    public List<FruitTransaction> convertToTransactions(List<String> csvLines) {
+        List<String> csvLinesCopy = csvLines;
+        csvLinesCopy.remove(0);
+        OperationHandlerSwitch operationHandlerSwitch = new OperationHandlerSwitchImpl();
+        List<FruitTransaction> listOfTransactions = new ArrayList<>();
+        for (int i = 0; i < csvLinesCopy.size(); i++) {
```

Comment:
The code to extract the type, fruit, and quantity looks repeated for each line, consider extracting this logic into a single method to make the code more concise and reduce redundancy. Try something like this:

`
    for (String csvLine : csvLinesCopy) {
        String[] parts = csvLine.split(",");
        if (parts.length != 3) {
            throw new IllegalArgumentException("CSV line format is incorrect: " + csvLine);
        }

        String type = parts[0].trim();
        String fruit = parts[1].trim();
        int quantity = parseQuantity(parts[2].trim(), csvLine);

        listOfTransactions.add(new FruitTransaction(
            operationHandlerSwitch.getOperation(type), fruit, quantity));
    }

private int parseQuantity(String quantityStr, String csvLine) {
    int quantity;
    try {
        quantity = Integer.parseInt(quantityStr);
    } catch (NumberFormatException e) {
        throw new IllegalArgumentException("Invalid quantity value in line: " + csvLine, e);
    }
    if (quantity < 0) {
        throw new IllegalArgumentException("Quantity cannot be less than 0, actual quantity: " + quantity);
    }
    return quantity;
}`